doc/Common_Practices_Typescript.md

```diff
@@ -0,0 +1,8 @@
+# Common practices for client-side code
+
+These guidelines are agreed-upon practices by the FUN team and reflect their current understanding of Vue.js and TypeScript. 
+
+* We collect all new types in the `types.ts` file. When it becomes sufficiently large, we'll have enough code "examples" to refactor them into smaller cohesive units/modules.
+* When writing a Vuex store module, don't split mutations, actions, getters, etc in separate files. Doing this makes the code harder to navigate. If a Vuex module file becomes too big, ask the team members if splitting is ok.
```

Comment:
We did that split. Beyond smaller files it also gives you to ability to import only the actionTypes for example in order to perform an action without having to load all of the other store stuff along with it.
Examples:
* https://gerrit.wikimedia.org/r/plugins/gitiles/wikibase/termbox/+/master/src/store/language/ (w/ store modules)
* https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/WikibaseLexeme/+/master/resources/store/ (a monostore)

Reply:
We took inspiration from your code and did not like the "split style". 

At the moment we don't have actionTypes (using plain strings), but as soon as we do, splitting them into separate files makes total sense.